subprojects/plugin-use/src/integTest/groovy/org/gradle/plugin/use/DeployedPortalIntegrationSpec.groovy

```diff
@@ -154,4 +154,41 @@ class DeployedPortalIntegrationSpec extends AbstractPluginIntegrationTest {
         then:
         failure.assertThatDescription(startsWith("Plugin [id: '$HELLO_WORLD_PLUGIN_ID', version: '$HELLO_WORLD_PLUGIN_VERSION'] was not found"))
     }
+
+    def "can resolve plugin from portal with repository filters present"() {
+        given:
+        def helloWorldGroup = 'org.gradle'
+        def helloWorldName = 'gradle-hello-world-plugin'
+
+        when:
+        buildScript """
+            buildscript {
+                repositories {
+                    exclusiveContent {
+                      forRepository {
+                        maven {
+                          url = "https://storage.googleapis.com/r8-releases/raw"
+                          metadataSources {
+                            artifact()
+                          }
+                        }
+                      }
+                      filter {
+                        includeModule("com.android.tools", "r8")
+                      }
+                    }
+                }
+            }
+
+            plugins {
+                id "$HELLO_WORLD_PLUGIN_ID" version "$HELLO_WORLD_PLUGIN_VERSION"
+            }
+        """
+
+        then:
+        succeeds("helloWorld")
```

Comment:
:x: I would probably add a test case where resolution fails, otherwise you don't really prove the filter is applied.

Reply:
Added another test that fails resolution enforced by the content filters. Please let me know if there are better ways to write such tests without hitting maven central.

Reply:
I have example tests for locking that deal with buildscript classpath dependencies and do not require a remote hit. Have a look at `UsingLockingOnNonProjectConfigurationsIntegrationTest` and we can discuss if needed.

Reply:
Thanks!